azurelinuxagent/common/protocol/wire.py

```diff
@@ -858,47 +859,56 @@ def check_wire_protocol_version(self):
             raise ProtocolNotFoundError(error)
 
     def _call_hostplugin_with_container_check(self, host_func):
-        ret = None
+        """
+        Calls host_func on host channel and accounts for stale resource (ResourceGoneError or InvalidContainerError).
+        If stale, it refreshes the goal state and retries host_func.
+        This method can throw, so the callers need to handle that.
+        """
         try:
             ret = host_func()
-        except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
+            if ret is None or not ret:
```

Comment:
Oh, didn't notice this one until I had already commented the comment below.

Same thing, though.

Reply:
Same.